### PR TITLE
Onnx version update from 1.2.1 to 1.3 in CI

### LINKED ISSUE #12049: Flaky OnnxBackendRealModelTest.test_bvlc_alexnet_cpu - connection timeout
http://jenkins.mxnet-ci.amazon-ml.com/blue/organizations/jenkins/incubator-mxnet/detail/master/1330/pipeline
```
________________ OnnxBackendRealModelTest.test_bvlc_alexnet_cpu ________________



test_self = <mxnet_backend_test.OnnxBackendRealModelTest testMethod=test_bvlc_alexnet_cpu>

device = 'CPU'



    def run(test_self, device):  # type: (Any, Text) -> None

        if model_test.model_dir is None:

>           model_dir = self._prepare_model_data(model_test)



/usr/local/lib/python3.5/dist-packages/onnx/backend/test/runner/__init__.py:239: 

_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

/usr/local/lib/python3.5/dist-packages/onnx/backend/test/runner/__init__.py:182: in _prepare_model_data

    urlretrieve(model_test.url, download_file.name)

/usr/lib/python3.5/urllib/request.py:217: in urlretrieve

    block = fp.read(bs)

/usr/lib/python3.5/http/client.py:448: in read

    n = self.readinto(b)

/usr/lib/python3.5/http/client.py:488: in readinto

    n = self.fp.readinto(b)

/usr/lib/python3.5/socket.py:575: in readinto

    return self._sock.recv_into(b)

/usr/lib/python3.5/ssl.py:929: in recv_into

    return self.read(nbytes, buffer)

/usr/lib/python3.5/ssl.py:791: in read

    return self._sslobj.read(len, buffer)

_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 



self = <ssl.SSLObject object at 0x7fc9b0117908>, len = 4490

buffer = <memory at 0x7fc9b26a3288>



    def read(self, len=1024, buffer=None):

        """Read up to 'len' bytes from the SSL object and return them.

    

            If 'buffer' is provided, read into this buffer and return the number of

            bytes read.

            """

        if buffer is not None:

>           v = self._sslobj.read(len, buffer)

E           ConnectionResetError: [Errno 104] Connection reset by peer



/usr/lib/python3.5/ssl.py:575: ConnectionResetError

----------------------------- Captured stdout call -----------------------------

Start downloading model bvlc_alexnet from https://s3.amazonaws.com/download.onnx/models/opset_7/bvlc_alexnet.tar.gz

Failed to prepare data for model bvlc_alexnet: [Errno 104] Connection reset by peer

============= 1 failed, 189 passed, 656 skipped in 500.47 seconds ==============
```

### PR DESCRIPTION
**Pull request by Roshrini**
## Description ##
Fixing broken tests after upgrading ONNX 1.2.1 to ONNX 1.3

@vandanavk @zhreshold 
Fixes #12049

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

**Comment by stu1130** [comment]
Thanks for your contribution @Roshrini 
@mxnet-label-bot[pr-awaiting-review]

**Comment by vandanavk** [comment]
LGTM.

@anirudhacharya

**Comment by JimmyJamManson23** [comment]
For ducks skate get this to ivanka's sister asap. You can't go without me . I refuse to take the jet as it has already been touched. And please dont leave without me.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

File: python/mxnet/contrib/onnx/mx2onnx/_op_translations.py [hunk 1]
```diff
@@ -64,11 +64,11 @@
 def import_onnx_modules():
     """ To make sure ONNX is runtime dependency, it is imported used only when needed"""
     try:
-        from onnx import helper, numpy_helper, mapping
+        from onnx import helper, numpy_helper, mapping, TensorProto
```
Comment (anirudhacharya) [inline python/mxnet/contrib/onnx/mx2onnx/_op_translations.py]: Why do the import this way? Why not do this at the beginning along with other import statements. This sort of import creates a lot of other unnecessary changes in other methods.

File: python/mxnet/contrib/onnx/onnx2mx/_op_translations.py [hunk 2]
```diff
@@ -118,11 +118,23 @@ def add_n(attrs, inputs, proto_obj):
 # Sorting and Searching
 def argmax(attrs, inputs, proto_obj):
     """Returns indices of the maximum values along an axis"""
-    return 'argmax', attrs, inputs
+    new_attrs = translation_utils._fix_attribute_names(attrs, {'axis': 'axis',
```
Comment (anirudhacharya) [inline python/mxnet/contrib/onnx/onnx2mx/_op_translations.py]: what is this statement doing? it seems redundant.

File: python/mxnet/contrib/onnx/onnx2mx/_op_translations.py [hunk 3]
```diff
@@ -118,11 +118,23 @@ def add_n(attrs, inputs, proto_obj):
...
+                                                               'keepdims': 'keepdims'})
+    axis = new_attrs.get('axis', 0)
+    keepdims = new_attrs.get('keepdims', 0)
+    argmax_op = symbol.argmax(inputs[0], axis=axis, keepdims=keepdims)
+    cast_attrs = {'dtype': 'int64'}
```
Comment (anirudhacharya) [inline python/mxnet/contrib/onnx/onnx2mx/_op_translations.py]: why are we casting the result to int64?

File: python/mxnet/contrib/onnx/onnx2mx/_op_translations.py [hunk 4]
```diff
@@ -118,11 +118,23 @@ def add_n(attrs, inputs, proto_obj):
 # Sorting and Searching
 def argmax(attrs, inputs, proto_obj):
     """Returns indices of the maximum values along an axis"""
-    return 'argmax', attrs, inputs
+    new_attrs = translation_utils._fix_attribute_names(attrs, {'axis': 'axis',
+                                                               'keepdims': 'keepdims'})
+    axis = new_attrs.get('axis', 0)
+    keepdims = new_attrs.get('keepdims', 0)
```
Comment (anirudhacharya) [inline python/mxnet/contrib/onnx/onnx2mx/_op_translations.py]: the default value for keepdims is 1 not 0 - https://github.com/onnx/onnx/blob/master/docs/Operators.md#ArgMax

File: python/mxnet/contrib/onnx/onnx2mx/_op_translations.py [hunk 5]
```diff
@@ -118,11 +118,23 @@ def add_n(attrs, inputs, proto_obj):
...
+    argmax_op = symbol.argmax(inputs[0], axis=axis, keepdims=keepdims)
+    cast_attrs = {'dtype': 'int64'}
+    return 'cast', cast_attrs, argmax_op
 
 def argmin(attrs, inputs, proto_obj):
     """Returns indices of the minimum values along an axis."""
-    return 'argmin', attrs, inputs
+    new_attrs = translation_utils._fix_attribute_names(attrs, {'axis': 'axis',
```
Comment (anirudhacharya) [inline python/mxnet/contrib/onnx/onnx2mx/_op_translations.py]: same as above.

File: tests/python-pytest/onnx/export/onnx_backend_test.py [hunk 6]
```diff
@@ -80,7 +80,7 @@
     'test_reciprocal',
     'test_sqrt',
     'test_pow',
-    'test_exp',
+    'test_exp_example',
```
Comment (anirudhacharya) [inline tests/python-pytest/onnx/export/onnx_backend_test.py]: won't this end up excluding a few tests?

File: tests/python-pytest/onnx/import/mxnet_backend_test.py [hunk 7]
```diff
@@ -39,12 +39,12 @@
 for op_tests in test_cases.IMPLEMENTED_OPERATORS_TEST:
     BACKEND_TESTS.include(op_tests)
 
-for std_model_test in test_cases.STANDARD_MODEL:
-    BACKEND_TESTS.include(std_model_test)
-
 for basic_model_test in test_cases.BASIC_MODEL_TESTS:
     BACKEND_TESTS.include(basic_model_test)
 
+for std_model_test in test_cases.STANDARD_MODEL:
```
Comment (anirudhacharya) [inline tests/python-pytest/onnx/import/mxnet_backend_test.py]: what is the change here?

File: tests/python-pytest/onnx/export/onnx_backend_test.py (hunk 6 above)
Comment (Roshrini) [reply tests/python-pytest/onnx/export/onnx_backend_test.py]: 'test_exp' includes 'expand' operator tests too which is not implemented yet

File: tests/python-pytest/onnx/import/mxnet_backend_test.py (hunk 7 above)
Comment (Roshrini) [reply tests/python-pytest/onnx/import/mxnet_backend_test.py]: just sequence changed

File: python/mxnet/contrib/onnx/mx2onnx/_op_translations.py (hunk 1 above)
Comment (Roshrini) [reply python/mxnet/contrib/onnx/mx2onnx/_op_translations.py]: here functions are invoked according to the operators in the model.. so try catch with import statement doesn't work if you just do it at the top. You will have to import with try catch in every method, so this way seems much better

File: python/mxnet/contrib/onnx/onnx2mx/_op_translations.py (hunk 3 above)
Comment (Roshrini) [reply python/mxnet/contrib/onnx/onnx2mx/_op_translations.py]: argmax/argmin output is always int64 in onnx https://github.com/onnx/onnx/blob/master/docs/Operators.md#ArgMax
Tests will fail otherwise

File: tests/python-pytest/onnx/export/onnx_backend_test.py (hunk 6 above)
Comment (anirudhacharya) [reply tests/python-pytest/onnx/export/onnx_backend_test.py]: can you please use `test_exp_'?

Comment (Roshrini) [reply tests/python-pytest/onnx/export/onnx_backend_test.py]: That will also exclude 'test_exp' test. https://github.com/onnx/onnx/blob/master/docs/Operators.md#Exp

File: python/mxnet/contrib/onnx/mx2onnx/_op_translations.py (hunk 1 above)
Comment (anirudhacharya) [reply python/mxnet/contrib/onnx/mx2onnx/_op_translations.py]: TensorProto is used only in one method. Would it make sense to move the import of Tensorproto to the respective method?

File: tests/python-pytest/onnx/export/onnx_backend_test.py (hunk 6 above)
Comment (anirudhacharya) [reply tests/python-pytest/onnx/export/onnx_backend_test.py]: ONNX appends `_cpu` to all the node tests. So `test_exp` will be `test_exp_cpu` when it is run. Which is why I think using `test_exp_' might work?

File: python/mxnet/contrib/onnx/onnx2mx/_op_translations.py (hunk 2 above)
Comment (Roshrini) [reply python/mxnet/contrib/onnx/onnx2mx/_op_translations.py]: removed

File: python/mxnet/contrib/onnx/onnx2mx/_op_translations.py (hunk 4 above)
Comment (Roshrini) [reply python/mxnet/contrib/onnx/onnx2mx/_op_translations.py]: fixed

File: python/mxnet/contrib/onnx/onnx2mx/_op_translations.py (hunk 5 above)
Comment (Roshrini) [reply python/mxnet/contrib/onnx/onnx2mx/_op_translations.py]: removed

File: python/mxnet/contrib/onnx/mx2onnx/_op_translations.py (hunk 1 above)
Comment (Roshrini) [reply python/mxnet/contrib/onnx/mx2onnx/_op_translations.py]: Importing "onnx" in import_onnx_modules now instead of helper functions.

File: python/mxnet/contrib/onnx/mx2onnx/_op_translations.py [hunk 8]
```diff
@@ -1841,20 +1840,24 @@ def convert_cast(node, **kwargs):
...
     """
-    helper, _, _ = import_onnx_modules()
+    onnx = import_onnx_modules()
     name = node["name"]
     proc_nodes = kwargs["proc_nodes"]
     inputs = node["inputs"]
     dtype = node["attrs"]["dtype"]
+    if dtype == 'float32':
+        dtype = 'float'
+    elif dtype == 'float64':
```
Comment (sandeep-krishnamurthy) [inline python/mxnet/contrib/onnx/mx2onnx/_op_translations.py]: whats the behavior for other types and should we handle default type?

Comment (sandeep-krishnamurthy) [reply python/mxnet/contrib/onnx/mx2onnx/_op_translations.py]: I think it is better if we do type = numpy.dtype(dtype) and later type.name(). What do you think?

File: python/mxnet/contrib/onnx/onnx2mx/_op_translations.py (hunk 3 above)
Comment (sandeep-krishnamurthy) [reply python/mxnet/contrib/onnx/onnx2mx/_op_translations.py]: Please add it as code comment for easy maintenance.

File: python/mxnet/contrib/onnx/onnx2mx/_op_translations.py [hunk 9]
```diff
@@ -118,11 +118,19 @@ def add_n(attrs, inputs, proto_obj):
...
+    cast_attrs = {'dtype': 'int64'}
+    return 'cast', cast_attrs, argmax_op
 
 def argmin(attrs, inputs, proto_obj):
     """Returns indices of the minimum values along an axis."""
-    return 'argmin', attrs, inputs
+    axis = attrs.get('axis', 0)
+    keepdims = attrs.get('keepdims', 1)
+    argmin_op = symbol.argmin(inputs[0], axis=axis, keepdims=keepdims)
+    cast_attrs = {'dtype': 'int64'}
```
Comment (sandeep-krishnamurthy) [inline python/mxnet/contrib/onnx/onnx2mx/_op_translations.py]: same as above.

File: python/mxnet/contrib/onnx/mx2onnx/_op_translations.py (hunk 8 above)
Comment (sandeep-krishnamurthy) [reply python/mxnet/contrib/onnx/mx2onnx/_op_translations.py]: Please add comment on why float instead of float32 and double instead of float64.

**Comment by vandanavk** [comment]
Please add 

> Fixes https://github.com/apache/incubator-mxnet/issues/12049

 in the PR description. It will automatically close the issue when the PR is merged.

File: python/mxnet/contrib/onnx/onnx2mx/_op_translations.py (hunk 9 above)
Comment (Roshrini) [reply python/mxnet/contrib/onnx/onnx2mx/_op_translations.py]: added comments

File: python/mxnet/contrib/onnx/mx2onnx/_op_translations.py (hunk 8 above)
Comment (Roshrini) [reply python/mxnet/contrib/onnx/mx2onnx/_op_translations.py]: This is mapping from mxnet's dtype for cast operator to onnx's dtypes. Added comment to better explain.

TensorProto maps 'float32' to 'float' and 'float64' to 'double'. Rest of the types are handled by this line while creating cast op node->   " to=getattr(onnx.TensorProto, dtype.upper())",

**Comment by anirudhacharya** [comment]
LGTM

**Comment by vandanavk** [comment]
@sandeep-krishnamurthy Please change the label to pr-awaiting-review

**Comment by Roshrini** [comment]
@sandeep-krishnamurthy Can we merge this PR?

**Comment by sandeep-krishnamurthy** [comment]
@zhreshold - Can you take a look? Thanks.

**Comment by vandanavk** [comment]
@sandeep-krishnamurthy  This PR is ready for merge.
